include/web_video_server/image_streamer.h

```diff
@@ -62,6 +64,19 @@ class ImageTransportImageStreamer : public ImageStreamer
   virtual void restreamFrame(double max_age);
   virtual void initialize(const cv::Mat &);
 
+  /**
+   * @brief Detects QoS profile settings used by publishers on a specified topic.
+   * 
+   * This function queries the ROS2 middleware to discover publishers on the 
+   * given topic and extracts their QoS profile settings. It's useful for 
+   * creating subscribers that automatically match the publisher's QoS.
+   * 
+   * @param topic_name The full name of the topic to query.
+   * @return std::optional<rmw_qos_profile_t> The detected QoS profile if a publisher
```

Comment:
```suggestion
   * @return The detected QoS profile if a publisher
```
To me, repeating the return type in the doxygen comments is redundant as you can read it in source and actual doxygen will state it from the source code also.